src/library/cmd_reader.h

```diff
@@ -184,5 +163,160 @@ namespace xlang::cmd
         }
 
         std::map<std::string_view, std::vector<std::string>> m_options;
+
+		template<typename O, typename L>
+		void extract_option(std::string_view arg, O const& options, L& last)
+		{
+			if (arg[0] == '-')
+			{
+				arg.remove_prefix(1);
+				last = find(options, arg);
+
+				if (last == options.end())
+				{
+					throw_invalid("Option '-", arg, "' is not supported");
+				}
+
+				m_options.try_emplace(last->name);
+			}
+			else if (arg[0] == '@')
+			{
+				arg.remove_prefix(1);
+				extract_response_file(arg, options);
+			}
+			else if (last == options.end())
+			{
+				throw_invalid("Value '", arg, "' is not supported");
+			}
+			else
+			{
+				m_options[last->name].push_back(std::string{ arg });
+			}
+		}
+
+		template<typename O>
+		void extract_response_file(std::string_view const& arg, O const& options)
+		{
+			std::experimental::filesystem::path response_path{ std::string{ arg } };
+			std::string extension = response_path.extension().generic_string();
+			std::transform(extension.begin(), extension.end(), extension.begin(), tolower);
+			if (is_directory(response_path) || extension == ".winmd")
+			{
+				throw_invalid("'@' is reserved for response files");
+			}
+			std::array<char, 8192> line_buf;
+			std::ifstream response_file(absolute(response_path));
+			while (response_file.getline(line_buf.data(), line_buf.size()))
+			{
+				size_t argc = 0;
+				std::vector<std::string> argv;
+
+				parse_command_line(line_buf.data(), argv, &argc);
+
+				auto last{ options.end() };
+				for (auto i = 0; i < argc; i++)
+				{
+					extract_option(std::string_view{ argv[i] }, options, last);
+				}
+			}
+		}
+
+		template <typename Character>
+		static void parse_command_line(Character* cmdstart, std::vector<std::string>& argv, size_t* argument_count)
+		{
+			*argument_count  = 0;
+
+			Character c;
+			std::string arg("");
+			int copy_character; // 1 = copy char to arg
+			unsigned numslash; // num of backslashes seen
+			bool in_quotes;
+			bool first_arg;
+
+			Character* p = cmdstart;
+			in_quotes = false;
+			first_arg = true;
+
+			// Loop on each argument
```

Comment:
Please remove comments that don't substantially make the code more understandable.